pkg/graph/resolvers/apply_changes.go

```diff
@@ -0,0 +1,78 @@
+package resolvers
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/mitchellh/mapstructure"
+)
+
+// sanitizeChanges performs some pre-processing of the changes map for "known" cases that we'd like to handle.
+// Currently, these include:
+// - Empty strings are converted to nil
+// - Empty slices are converted to nil
+func sanitizeChanges(changes map[string]interface{}) {
+	for key, value := range changes {
+		// Get the reflect value for type comparisons
+		reflectValue := reflect.ValueOf(value)
+
+		// String operations
+		if reflectValue.Kind() == reflect.String {
+			valAsString, ok := reflectValue.Interface().(string)
+
+			// Convert empty strings to `nil`
+			if ok && len(valAsString) == 0 {
+				changes[key] = nil
+				continue
+			}
+		}
+
+		// Empty slices don't play well with mapstructure, as they enter as []interface{}
+		// which promptly gets ignored by mapstructure.
+		// In order to get around this, we'll convert empty slices to a real "nil" value
+		if reflectValue.Kind() == reflect.Slice && reflectValue.IsNil() {
+			changes[key] = nil
+		}
+	}
+}
+
+// ApplyChanges applies arbitrary changes from a map to a struct
```

Comment:
Might be helpful to expand this comment to mention that fields not mentioned in `changes` will stay the same.